modules/zookeeper/src/test/java/org/apache/ignite/spi/discovery/zk/internal/ZookeeperDiscoveryTopologyChangeAndReconnectTest.java

```diff
@@ -62,11 +66,35 @@
  * Tests for Zookeeper SPI discovery.
  */
 public class ZookeeperDiscoveryTopologyChangeAndReconnectTest extends ZookeeperDiscoverySpiTestBase {
+    /** {@code True} if indexing disabled. */
+    private boolean indexingDisabled;
+
     /** {@inheritDoc} */
     @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
-        return super.getConfiguration(igniteInstanceName).setIncludeEventTypes(EventType.EVTS_ALL);
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setIncludeEventTypes(EventType.EVTS_ALL);
+
+        if (indexingDisabled) {
+            GridQueryProcessor.idxCls = DummyQueryIndexing.class;
+
+            cfg.setSystemViewExporterSpi(new JmxSystemViewExporterSpi() {
+                @Override protected void register(SystemView<?> sysView) {
+                    // No-op.
+                }
+            });
+        }
+
+        return cfg;
     }
-    
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        indexingDisabled = false;
```

Comment:
You should also return the value of `GridQueryProcessor.idxCls` to `null` (in case startGrid fails)

Reply:
Done.